### PR TITLE
test: Fix drift testing with AL2023 on 1.23

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

Fix drift testing with AL2023 on 1.23. During our version compatability testing, we are trying to get an AMI that is a minor version older than the current minor version of K8s that we are working with. This works well for every version except for 1.23, because AL2023 doesn't have support in 1.22. Thus, we are just pinning to an older version of AL2023 on 1.23 when testing on that version.

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: test/suites/integration/ami_test.go [hunk 1]
```diff
@@ -156,7 +156,7 @@ var _ = Describe("AMI", func() {
 		It("should provision a node using the Ubuntu family", func() {
 			nodeClass.Spec.AMIFamily = &v1beta1.AMIFamilyUbuntu
 			// TODO (jmdeal@): remove once 22.04 AMIs are supported
-			if env.GetK8sVersion(0) == "1.29" {
+			if env.K8sMinorVersion() >= 29 {
 				nodeClass.Spec.AMISelectorTerms = lo.Map([]string{
```
Comment (jmdeal) [inline test/suites/integration/ami_test.go:160]: nit: also update this to use env.GetAMIBySSMPath?

**Review by jmdeal — DISMISSED** [review]
One nit but LGTM otherwise 🚀

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — APPROVED** [review]
LGTM 🚀

**Pull request merged**